requirements-ci.txt

```diff
@@ -1,4 +1,4 @@
-gensim<4.0.0
+gensim>=4.0.0
```

Comment:
```suggestion
gensim
```

Reply:
`gensim.doctest` will require 4.0.0 or higher, but I suppose that `requirements-ci.txt` will take the highest version regardless. I'm not sure whether it's preferable to have `gensim>=4.0.0`, where all valid versions will pass the tests, or just `gensim` which is simpler, but the tests would fail for `gensim<4.0.0`.

Also, what are your thoughts on removing `gensim` from `pip-req.txt`?

Reply:
I don't mind either way, just a thought.

As you said, CI will use the latest version and invalidate the cache when this file changes. So we'll always have a working version until the next breaking change.

Onboard with trimming down `pip-req.txt` to remove things that we don't directly need for features. We have a CI pipeline to cover the doctest so unless someone is working on that specific file, they won't need the dependency in local dev environment, which is what we use the `pip-req.txt` file for according to [docs](https://github.com/nltk/nltk/blob/develop/CONTRIBUTING.md#setting-up-a-development-environment).

Reply:
I'll keep it as `gensim>=4.0.0` solely as an indication that the tests in the CI only pass for those versions. Once we have this PR merged or closed, I'll move to trimming `pip-req.txt`, renaming it to `requirements.txt`, and deleting either `requirements-ci.txt` and `requirements-test.txt`.